packages/devtools_app/lib/src/debugger/flutter/debugger_screen.dart

```diff
@@ -32,66 +36,505 @@ class DebuggerScreenBody extends StatefulWidget {
   DebuggerScreenBodyState createState() => DebuggerScreenBodyState();
 }
 
-class DebuggerScreenBodyState extends State<DebuggerScreenBody> {
+class DebuggerScreenBodyState extends State<DebuggerScreenBody>
+    with AutoDisposeMixin {
+  DebuggerController controller;
+  ScriptRef loadingScript;
   Script script;
+  ScriptList scriptList;
+  Stack stack;
 
   @override
   void initState() {
     super.initState();
-    // TODO(https://github.com/flutter/devtools/issues/1648): Make file picker.
-    // Make the loading process disposable.
+    // TODO(djshuckerow): promote this controller to app-level Controllers.
+    controller = DebuggerController();
+    controller.setVmService(serviceManager.service);
+    addAutoDisposeListener(controller.isPaused, () async {
+      print('Pause status changed');
+      // TODO(https://github.com/flutter/devtools/issues/1648): Allow choice of
+      // the scripts on the stack.
+      if (controller.isPaused.value) {
+        print('Paused is true, updating stack');
+        final currentStack = await controller.getStack();
+        // NOT READY YET: I don't understand why the getStack() call is returning null here.
```

Comment:
That's odd - what json result are you getting back from the VM? It's possible you're getting a result, but the vm service protocol library is having trouble converting that to dart objects.

You may also want to print out the isolate id you're requesting the stack for.

Reply:
It appears to have been a Sentinel object that I was getting.

The HTML code has provisions for this situation that I had not used.